caddy.go

```diff
@@ -877,3 +877,7 @@ type CtxKey string
 // context.WithValue to access the original request URI that accompanied the
 // server request. The associated value will be of type string.
 const URLPathCtxKey CtxKey = "url_path"
+
+// URIxRewriteCtxKey is a context key used to store original unrewritten
+// URI in context.WithValue
+const URIxRewriteCtxKey CtxKey = "caddy_rewrite_original_uri"
```

Comment:
Now that I see URLPathCtxKey and URIxRewriteCtxKey constants, I'm beginning to think they won't necessarily be used outside of the httpserver and am thinking these should be declared in the `httpserver` package instead.